lib/utils/parseAstCode.ts

```diff
@@ -0,0 +1,63 @@
+import mergeWith from "lodash.mergewith";
+import type { ISbStoriesParams, StoryblokBridgeConfigV2 } from "@storyblok/js";
+import type { RawCodeOptions } from "../vite-plugins/vite-plugin-storyblok-bridge";
+import type { Rollup } from "vite";
+import type { AwaitExpression, SpreadElement, Property } from "estree";
+
+/**
+ * Parses through the Abstract Syntax Tree (AST) code to locate the 'useStoryblok' function and its properties.
+ * This functionality is crucial for generating a virtual module that can be utilized during the initialization of the Storyblok bridge.
+ */
+
+export function parseAstRawCode(astCode: Rollup.ProgramNode) {
+  let obj: RawCodeOptions = {};
+
+  function customizer(_: any, srcValue: AwaitExpression) {
+    if (
+      srcValue?.type === "AwaitExpression" &&
+      srcValue.argument.type === "CallExpression" &&
+      srcValue.argument.callee.type === "Identifier" &&
+      srcValue.argument.callee.name === "useStoryblok"
+    ) {
+      const props = srcValue?.argument?.arguments;
+      if (props && props[1]?.type === "ObjectExpression") {
+        const apiOptions = getAstPropToObj(props[1]?.properties);
+        obj = {
+          ...obj,
+          apiOptions,
+        };
+      }
+      if (props && props[2]?.type === "ObjectExpression") {
+        const bridgeOptions = getAstPropToObj(props[2]?.properties);
+        obj = {
+          ...obj,
+          bridgeOptions,
+        };
+      }
+    }
+  }
+
+  mergeWith({}, astCode, customizer);
+  return obj;
+}
+
+function getAstPropToObj(properties: (SpreadElement | Property)[]) {
+  const option: ISbStoriesParams | StoryblokBridgeConfigV2 = {};
+
+  return properties.reduce((options, property) => {
+    if (property.type !== "Property") return options;
+    const { key, value } = property;
+    const { type } = value;
+    if (key.type !== "Identifier") return options;
+
+    if (type === "Literal") {
+      options[key.name] = value.value;
+    } else if (type === "ArrayExpression") {
+      options[key.name] = value.elements
+        .filter((v) => v.type === "Literal")
```

Comment:
I get a TS error here because TS cannot infer types from the `filter` function and we have to provide some kind of type guard to narrow the type.

```
(v): v is Literal => v.type === "Literal"
```
`Literal` is also exported from `estree`

Reply:
That's perfect 👌 
At the end of the day, we probably don't need to enforce that type tho, as those values are going to be provided to our `ISbStoriesParams | StoryblokBridgeConfigV2`. I'd just use a `reducer` to get rid of the "filter + map" TS error. Something like:

```
      const arrayValues = value.elements.reduce((acc, element) => {
        if (element.type === "Literal" && element.value) {
          return [...acc, element.value];
        }
        return acc;
      }, []);

      options[key.name] = arrayValues;
```

WDYT?